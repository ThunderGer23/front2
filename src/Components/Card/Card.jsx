import PropTypes from 'prop-types'

export const Card = ({titulo, texto}) => {
  return (
    <div role='div' className="card-body">
        <h3 role='h3'>{titulo}</h3>
        <p role='p'>{texto}</p>
    </div>
  )
}

Card.propTypes = {
    titulo: PropTypes.string.isRequired,
    texto: PropTypes.string.isRequired
}