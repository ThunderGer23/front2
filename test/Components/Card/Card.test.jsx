import { render, screen } from "@testing-library/react";
import { Card } from "../../../src/Components/Card/Card";

describe('Pruebas en el componente Card', () => { 
    const titulo = 'Titulo Card'
    const texto = 'Texto del Card'

    test('Esta prueba es para hacer match con el snapshot', () => { 
        const {container} = render(<Card titulo={titulo} texto={texto}/>)//Renderiza el componente Card
        expect(container).toMatchSnapshot();//Toma una foto
     })

    test('Esta prueba valida que el div exista y pertenezca a la clase correcta', () => { 
        render(<Card titulo={titulo} texto={texto}/>)
        expect(screen.getByRole('div')).toBeTruthy()//Se valida que exista el div
        const {className} = screen.getByRole('div')//Se salva la clase del div del componente en className
        expect(className).toBe('card-body')//Se valida que el div tenga la misma clase que className
     })

    test('Esta prueba valida que la etiqueta h3 exista', () => { 
        render(<Card titulo={titulo} texto={texto}/>)
        expect(screen.getByRole('h3')).toBeTruthy()
     })

    test('Esta prueba valida que la etiqueta p exista', () => { 
        render(<Card titulo={titulo} texto={texto}/>)
        expect(screen.getByRole('p')).toBeTruthy()
     })
 })