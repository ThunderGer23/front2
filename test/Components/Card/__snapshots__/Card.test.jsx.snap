// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente Card Esta prueba es para hacer match con el snapshot 1`] = `
<div>
  <div
    class="card-body"
    role="div"
  >
    <h3
      role="h3"
    >
      Titulo Card
    </h3>
    <p
      role="p"
    >
      Texto del Card
    </p>
  </div>
</div>
`;
